### PR TITLE
Remove old Symfony 2.3 code

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targetting this branch, because this is removing compatibility with Symfony 2.3.

Closes #1057 

## Changelog

```markdown
### Changed
- Changed `MediaController::downloadAction` signature to include Request as first parameter
- Changed `MediaController::viewAction` signature to include Request as first parameter
- Changed `MediaController::liipImagineFilterAction` signature to include Request as first parameter
```

## Subject

I am removing unused lines of codes, used for symfony 2.3 compatibility on 3.x branch.
Also changing some controllers signatures to accept `$request`as a parameter.

### CODE REVIEW

File: Controller/MediaController.php [hunk 1]
```diff
@@ -44,27 +45,28 @@ public function getMedia($id)
...
-     * @param string $id
-     * @param string $format
+     * @param Request $request
+     * @param string  $id
+     * @param string  $format
      *
      * @return Response
      */
-    public function downloadAction($id, $format = 'reference')
+    public function downloadAction(Request $request, $id, $format = 'reference')
```
Comment (greg0ire) [inline Controller/MediaController.php]: This is a BC-break, which means there should be something in the `UPGRADE` guide at least, and at most, a PR should be made on the stable branch to simulate both signatures, and deprecate the old one (that would help with migrating calls, but not with migrating child classes).

Comment (jordisala1991) [inline Controller/MediaController.php]: Is there a way to simulate this signature on 3.x? I am not sure how to trigger this deprecation

Comment (jordisala1991) [inline Controller/MediaController.php]: I did update this PR with a note on UPGRADE-4.0 and a changelog. If we find a way to do this in 3.x, I will create another PR first for 3.x branch

Comment (greg0ire) [inline Controller/MediaController.php]: You can simulate any signature with `func_get_args()`

Comment (jordisala1991) [inline Controller/MediaController.php]: Do you mean like removing all parameters and get them using `func_get_args()`? The only problem I see with that is the fact that AFAIK Symfony only sends the request if you type hint that parameter.

Comment (greg0ire) [inline Controller/MediaController.php]: I meant by keeping the old signature, and check the first parameter's type, but indeed, I overlooked the fact that this is an action and some magic is happening here. So this isn't possible.

File: UPGRADE-4.0.md [hunk 2]
```diff
@@ -9,20 +9,25 @@ Please read [3.x](https://github.com/sonata-project/SonataMediaBundle/tree/3.x)
 
 See also the [diff code](https://github.com/sonata-project/SonataMediaBundle/compare/3.x...4.0.0).
 
+## Controllers
+
+`MediaController` action were changed to introduce `Request $request` as first parameter. You must update those signatures if you want to still extend this class: `downloadAction`, `listAction` and `liipImagineFilterAction`
```
Comment (greg0ire) [inline UPGRADE-4.0.md]: "actions"

Comment (greg0ire) [inline UPGRADE-4.0.md]: Please wrap to 120 chars, cutting your sentences in a semantical way.

**Comment by jordisala1991** [comment]
@sonata-project/contributors Please, review again

**Comment by OskarStark** [comment]
Great job @jordisala1991 👍 🎉

**Pull request merged**